### PR TITLE
Print view improvements

### PR DESCRIPTION
**Pull request by satwo**
WIP. EDIT: Ready for review.
 
- Adds a print button
- Modifies the warning message on the `Print` PoS view to include a link to the proper settings page (h/t @bolatovumar)
- Adds the css class `d-print-none` to hide some potentially undesired elements for the printed page (Store title, store description, and item inventory probably aren't appropriate for a printed page displayed in a storefront*). This also simplifies the formatting of the printed page.

Despite the last bullet point, I'm still running into trouble with the formatting of the bootstrap `card` elements on the page-to-print:

image

I also haven't figured out a way to prevent the items from going over the page break. I'm using the default 8.5x11 inch ANSI letter, but I assume we'd have similar issues with A4.

Paging @dstrukt @dennisreimann, any ideas?

*It would be nice to hear some feedback on this from users who plan to use the `print` PoS view, i.e.: what item elements do they need on the printed page? Presumably for formatting purposes, the fewer, the better (but again I'll defer to the CSS experts on this one).

### CODE REVIEW

File: BTCPayServer/wwwroot/main/site.css [hunk 1]
```diff
@@ -536,3 +536,15 @@ header.masthead .header-content .header-content-inner p {
...
+@media print {
+    @page {
+        margin-top: 0;
+        margin-bottom: 0;
+    }
+    body {
+        padding-top: 72px;
+        padding-bottom: 72px ;
+    }
+}
```
Comment (satwo) [inline BTCPayServer/wwwroot/main/site.css:550]: Proof-of-concept for hiding URL, datetime, and other metadata which some browsers add to a printed page. It probably shouldn't be here in site.css (or should be done differently to not affect all of BTCPay Server) - maybe the front-end folks can help find a better solution for this, too?

**Comment by dstrukt** [comment]
Will take a look and offer some suggestions, if I have any, when I get a moment later today.

**Comment by dennisreimann** [comment]
> I also haven't figured out a way to prevent the items from going over the page break.

Have you tried something like this?

```css
@media (print) {
  .card {
    page-break-inside: avoid;
  }
}
```

**Comment by satwo** [comment]
@dennisreimann Tested this with a simplied UI, no luck:

image

**Comment by dennisreimann** [comment]
Ok, I tried various things -- first conclusion: Chrome sucks in that regard and Firefox does a better job.

As we are using a separate template for the print view anyways, we can strip it of everything fancy for displaying the cards and solely focus on a good print outcome. Chrome has mainly issues with grid and flex display in that regard, so we need to unset this and align the cards the old school way. This worked well enough for me and I'd throw all of these one-off solutions directly into this particular view:

```html
<style>
    .card-deck { display: unset; }
    .card { page-break-inside: avoid; width: 300px; margin: 0 1em 1em 0; float: left; }
</style>
```

**Comment by satwo** [comment]
@dennisreimann Nice! That seemed to do the trick:

image

Still could use some tweaks - the non-print view of the `print` page is now slightly off-center, but I think that's a fine compromise to ensure the actual print functionality works properly. We can always improve it later, but at least now it should be usable for the intended purpose.

**Review by dennisreimann — APPROVED** [review]

**Comment by dennisreimann** [comment]
tACK. Streamlined it a bit more with some minor tweaks.

pos-print

**Pull request merged**